l2metrics/average_performance.py

```diff
@@ -0,0 +1,137 @@
+
+THE SOFTWARE IS PROVIDED “AS IS”, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR 
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, 
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE 
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, 
+WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR 
+IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+"""
+
+import logging
+
+import numpy as np
+import pandas as pd
+
+from ._localutil import fill_metrics_df
+from .core import Metric
+
+logger = logging.getLogger(__name__)
+
+
+class AvgPerf(Metric):
+    name = "Average Performance"
+    capability = "adapt_to_new_tasks"
+    requires = {"syllabus_type": "agent"}
+    description = (
+        "Calculates the average performance over all blocks,"
+        "effectively computing an area under the performance curve for training blocks"
+    )
+
+    def __init__(self, perf_measure: str, method: str = "mean") -> None:
+        super().__init__()
+        self.perf_measure = perf_measure
+
+        # Check for valid method
+        if method not in ["mean", "median"]:
+            raise KeyError(f"Invalid averaging method: {method}")
+        else:
+            self.do_mean = method == "mean"
+            self.do_median = method == "median"
+
+    def validate(self, block_info: pd.DataFrame) -> None:
+        # Initialize variables for checking block type format
+        last_block_num = -1
+        last_block_type = ""
+
+        # Ensure alternating block types
+        for _, regime in block_info.iterrows():
+            if regime["block_num"] != last_block_num:
+                last_block_num = regime["block_num"]
+
+                if regime["block_type"] == "test":
+                    if last_block_type == "test":
+                        logger.warning("Block types are not alternating")
+                        break
+                    last_block_type = "test"
+                elif regime["block_type"] == "train":
+                    if last_block_type == "train":
+                        logger.warning("Block types are not alternating")
+                        break
+                    last_block_type = "train"
+
+    def calculate(
+        self,
+        dataframe: pd.DataFrame,
+        block_info: pd.DataFrame,
+        metrics_df: pd.DataFrame,
+    ) -> pd.DataFrame:
+        # Validate block structure
+        self.validate(block_info)
+
+        # Use sleep evaluation blocks if they exist
+        if "sleep" in block_info["block_subtype"].to_numpy():
+            block_info_df = block_info[
+                ~(
+                    block_info.block_type.isin(["test"])
+                    & block_info.block_subtype.isin(["wake"])
+                )
+            ]
+        else:
+            block_info_df = block_info.copy()
+
+        # Set reward column
+        reward_col = "reward"
```

Comment:
This doesn't need to be hard-coded here. Should probably use self.perf_measure set in the constructor